src/consensus/aft/raft.h

```diff
@@ -440,7 +456,23 @@ namespace aft
       {
         guard.lock();
       }
+
       // This should only be called when the spin lock is held.
+
+      if (
+        !configurations.empty() &&
+        configurations.back().nodes.find(state->my_node_id) !=
+          configurations.back().nodes.end() &&
+        conf.find(state->my_node_id) == conf.end())
```

Comment:
This check deserves a comment, explaining that we're checking the node was present in a previous configuration before setting the retirement index (and ignoring any prefix of configurations before the node was ever added).

Reply:
Will add